e2e/main.py

```diff
@@ -157,5 +161,152 @@ def call_checker_endpoint(name, param, amount=0):
+                bulks[i : i + chunk_size] for i in range(0, len(bulks), chunk_size)
+            ]
+            for chunk_no, chunk in enumerate(chunks, 1):
+                print(
+                    "Sending",
+                    len(bulks),
+                    "operations as bulk:",
+                    "Chunk",
+                    chunk_no,
+                    "of",
+                    len(chunks),
+                )
+                inject(
+                    self.client,
+                    self.client.bulk(*chunk).autofill(ttl=MAX_OPERATIONS_TTL).sign(),
+                )
+
+        call_bulk(
+            [
+                checker.create_burrow((0, None)).with_amount(200_000_000),
+                checker.mint_kit((0, 80_000_000)),
+            ],
+            chunk_size=10,
+        )
+
+        call_endpoint(
+            ctez["ctez"], "create", (1, None, {"any": None}), amount=2_000_000
+        )
+        call_endpoint(ctez["ctez"], "mint_or_burn", (1, 100_000))
+        call_endpoint(ctez["fa12_ctez"], "approve", (checker.context.address, 100_000))
+
+        call_endpoint(
+            checker,
+            "add_liquidity",
+            (100_000, 100_000, 5, int(datetime.now().timestamp()) + 20),
+        )
+
+        burrows = list(range(1, 1001))
+
+        call_bulk(
+            [
+                checker.create_burrow((burrow_id, None)).with_amount(100_000_000)
+                for burrow_id in burrows
+            ],
+            chunk_size=100,
+        )
+
+        # Mint as much as possible from the burrows. All should be identical, so we just query the
+        # first burrow and mint that much kit from all of them.
+        max_mintable_kit = checker.metadata.burrowMaxMintableKit(
+            (self.client.key.public_key_hash(), 1)
+        ).storage_view()
+
+        bulks = []
+        for burrow_no in burrows:
+            bulks.append(
+                checker.touch_burrow((self.client.key.public_key_hash(), burrow_no))
+            )
+            bulks.append(checker.mint_kit(burrow_no, max_mintable_kit))
+
+        call_bulk(bulks, chunk_size=100)
+
+        # Change the index (kits are 100x valuable)
+        #
+        # Keep in mind that we're using a patched checker on tests where the protected index
+        # is much faster to update.
+        call_endpoint(oracle, "update", 100_000_000)
+
+        # Oracle updates lag one touch on checker
+        call_endpoint(checker, "touch", None)
+        call_endpoint(checker, "touch", None)
+        time.sleep(20)
+        call_endpoint(checker, "touch", None)
+
+        # Now burrows should be overburrowed, so we liquidate them all.
+        #
+        # This should use the push_back method of the AVL tree.
+        bulks = []
+        for burrow_no in burrows:
+            bid = (self.client.key.public_key_hash(), burrow_no)
+            bulks.append(checker.touch_burrow(bid))
+            bulks.append(checker.mark_for_liquidation(bid))
```

Comment:
We might be able to get rid of this call to `touch_burrow` as well, since `mark_for_liquidation` also implicitly calls `touch_burrow`.
```suggestion
            burrow_id = (self.client.key.public_key_hash(), burrow_no)
            bulks.append(checker.mark_for_liquidation(burrow_id))
```